### PR TITLE
Feature/399888 study period year logic change

### PR DESCRIPTION
**Pull request by PritiPiyushB**
### What one thing this PR does?

Study Period Filter - Implement Year Logic Changes

### Task details

- [399888 - Study Period Filter - Implement Year Logic Changes](https://dev.azure.com/defragovuk/DEFRA-NCEA/_workitems/edit/399888/)

### Other notes

- Removed ncea_catalogue_date from governence tab

### How do these changes look like?

Screenshot 2024-06-21 at 15 40 21

### Dev-tested in

1. Local environment

- [Search page](http://localhost:3000/search?q=lobster&jry=qs&pg=1&rpp=20&srt=best_match&sy=1900&ty=1905)

2. Dev environment

- [Search page](https://dev-ncea-search.azure.defra.cloud/search?q=lobster&jry=qs&pg=1&rpp=20&srt=best_match&sy=1900&ty=1906)
- ...

### CODE REVIEW

File: src/utils/formatSearchResponse.ts [hunk 1]
```diff
@@ -136,7 +136,6 @@ const getOtherDetails = async (
     metadata_standard: searchItem?._source?.standardNameObject?.default ?? '',
     project_number: projectId ?? '',
     metadata_language: searchItem?._source?.mainLanguage ?? '',
-    ncea_catalogue_date: formatDate(searchItem?._source?.dateStamp, false, false, '-'),
```
Comment (karthigayayiniram) [inline src/utils/formatSearchResponse.ts:139]: Please ensure that this is removed from anywhere it has been used.

Comment (PritiPiyushB) [reply src/utils/formatSearchResponse.ts:139]: Done

File: src/utils/queryBuilder.ts [hunk 2]
```diff
@@ -52,24 +52,41 @@ const _generateRangeBlock = (fields: IDateValues): estypes.QueryDslQueryContaine
...
+                { range: { 'resourceTemporalExtentDetails.end.date': { lt: toDateValue } } },
+              ],
+            },
+          },
+          {
+            bool: {
+              must: [
+                { range: { 'resourceTemporalExtentDetails.start.date': { gt: startDateValue } } },
+                { range: { 'resourceTemporalExtentDetails.end.date': { lt: toDateValue } } },
+              ],
```
Comment (karthigayayiniram) [inline src/utils/queryBuilder.ts:81]: As we discussed this query needs modification to cover all the scenarios specified on the AC. Please change the query accordingly

Comment (PritiPiyushB) [reply src/utils/queryBuilder.ts]: Resolved

**Review by karthigayayiniram — APPROVED** [review]

**Review by ganesh-rajamahanthy — APPROVED** [review]

**Pull request merged**